### PR TITLE
Port build system to Meson

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

Now that NumPy and SciPy have spent some time on Meson, it's not too bad to move our build system over to Meson as well. This PR also implements the move to `pyproject.toml` from #23829. I have added some documentation with some key points for migration, though most are edge cases that most developers shouldn't have to worry about.

The main change is that editable installs should disable build isolation:
```
$ pip install --no-build-isolation --editable .
```
This is because `pyproject.toml`-based install would normally isolate builds and extensions are built in the isolated environment, which goes against the idea of an editable install. (You will also need to install build requirements in your target environment, but I think that's not new anyway.)

The other downside is that all files need to be listed manually in most `meson.build`. _However_, this also has a side benefit of not accidentally installing extra files (e.g., I get several temporary `*.gcno` files and the whole `node_modules` directory if they happen to exist on `main`.)

The upside is that compilation is written in a very standard way. Both FreeType and Qhull are subprojects using manually written `meson.build` but those are just the same as the files in the main source. Qhull's build is a simple port of what we did in `setupext.py` to `meson.build`. FreeType's build is based on [the 2.9.1 version in Meson's wrapdb](https://wrapdb.mesonbuild.com), but dropping some of the extra install bits we won't use and rolled back to 2.6.1. For 2.11.1, we use the Meson build system already part of FreeType. When we move forward on FreeType versions, we can drop the patched-in build and use the upstream build system.

The other upside of Meson+meson-python builds is that compilation is a) faster with better parallelism since the entire set of files can be seen by the build system, and b) can be done _at import time_ in an editable install. The latter means that you will no longer need to re-install when editing C(++) files (unless you switch to an older commit.)

Closes #23829

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
And looking back at #23829, I forgot about adding the `dev` extra.

File: .github/workflows/cygwin.yml [hunk 1]
```diff
@@ -216,7 +216,7 @@ jobs:
           cat mplsetup.cfg
           # All dependencies must have been pre-installed, so that the minver
           # constraints are held.
-          python -m pip install --no-deps -ve .
+          python -m pip install --no-deps -v --no-build-isolation --ediable .
```
Comment (WeatherGod) [inline .github/workflows/cygwin.yml:219]: typo `--ediable` --> `--editable`

File: tools/generate_matplotlibrc.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+decommenting, which allows _auto_backend_sentinel to be filled in at import
+time.
+"""
+
+import sys
+from pathlib import Path
+
+
+if len(sys.argv) < 4:
+    raise SystemExit('usage: {sys.argv[0]} <input> <output> <backend>')
```
Comment (WeatherGod) [inline tools/generate_matplotlibrc.py:16]: This should have an `if __name__ == '__main__':` guard

File: .github/workflows/cygwin.yml (hunk 1 above)
Comment (QuLogic) [reply .github/workflows/cygwin.yml]: Yes, also missed some of the configuration options.

**Comment by QuLogic** [comment]
OK, I think we're at a good point now. A couple of issues had to be cleared up:
- I accidentally forgot to install the `macosx` backend, which made it unavailable in tests, so auto-picking in subprocesses chose TkAgg, which is still broken on Azure. I have fixed the backend here, but also made the subprocess explicitly use Agg where possible in #26680. Unfortunately, I don't think we can fix `test_lazy_auto_backend_selection` as its point is to not have a backend set.
- In some CI, we use the backend override from `mplsetup.cfg`; I've replicated that using the Meson equivalent, but I wonder if we should do that everywhere.
- Azure and AppVeyor have a ton of compilers preinstalled in addition to MSVC. Meson picks the non-MSVC option by default, so I have explicitly passed the option to use MSVC in CI. Since we don't specifically hard-require MSVC, I have not set that option in our config, and I expect most developers do not have multiple compilers on Windows, but it is a gotcha on CI.

There are also a couple followups that need to be considered eventually:
- Versions are not dynamically generated at build time. At runtime, we use `setuptools-scm`, so it is fine for editable installs. However, this does mean that wheels all use `3.9.0.dev0` as version, which reduces utility of the nightly wheels slightly. I'm not sure if we can hack around this post-build. Upstream is working on a better API for this: https://github.com/mesonbuild/meson-python/issues/159
- I don't think all license files are installed the same way as setuptools. I'm not sure if we have any easy way to fix that right now. There is an uptsream issue for this as well: https://github.com/mesonbuild/meson-python/issues/270

There isn't a huge change for developers, and you might be able to glean that from the CI changes (though most changes are extraneous for local work.) There may be a bit more change for distributions, but they might be used to that from NumPy/SciPy's existing change. I will write up some docs for that next week.

**Comment by QuLogic** [comment]
OK, there is actually one stall on the macOS arm64 builds, which I've reported upstream at https://github.com/mesonbuild/meson-python/issues/468

**Comment by ksunden** [comment]
I have found a solution that appears to _work_ regarding getting stubtest running with editable install, though not sure its actually a _good_ solution...

If you add the `$REPO_ROOT/lib` to the mesonpy-generated `matplotlib-editable.pth` file in `site-packages`, stubtest is able to run and give accurate results. (One slight change from setuptools editable install is that we had unused allowlist entries for untyped subpackages, but that is pretty trivial)

As far as I can tell it does not affect the special handling set up by mesonpy in `_matplotlib_editable_loader`, but does allow mypy to connect the stubs to the runtime.


All that said, while it may be good enough for a controlled CI environment, I don't like that as a solution for developers who may wish to run stubtest on their machine (note that `mypy` alone will work fine, just not with `-p` or `-m`, you must give it the `lib/matplotlib` path... stubtest requires runtime, so is more restricted) In my testing my editor integration still worked fine without any changes.

Perhaps this is a safe enough change and we should lobby/PR upstream to include it to support the mypy usecase.

The other option which seems to also be broken right now, but I don't think should be a problem, is to do a non-editable install... this is not great for the developer machine use case either, but at least does not involve editing generated files...

Currently I'm getting:

```
error: matplotlib failed to import, InvalidVersion: Invalid version: '"3.9.0.dev0"'
```

Which is a weird error on a couple fronts, firstly being why is version needed for importing in the first place?, secondly being why is that an invalid version?

Investigation has revealed that stubtest is doing (the `__import__` equivalent of) `from matplotlib import *`, and that is what is triggering the "InvalidVersion". Testing on a REPL confirms that while `import matplotlib` is fine, trying to get `__version_info__` (including a `*` import) breaks. The problem appears to be that there are quotes on `__version__`.


In summary:
   - I think after sorting out `__version_info__` problems probably the best short term path for CI is to use non-editable install for the stubtest job
   - If that doesn't work, editing the `pth` file should
   - May be worth talking to upstream about this usecase and seeing if the addition to the pth file can be included there, which would help for developer machines without having to edit a generated file.

**Comment by QuLogic** [comment]
> I think after sorting out `__version_info__` problems probably the best short term path for CI is to use non-editable install for the stubtest job

On editable installs, we trigger version calculation via `setuptools-scm`, whereas normal installs use the generated file from Meson. I thought I tested that. Apparently not, though.

**Comment by oscargus** [comment]
> `'"3.9.0.dev0"'`

Double quotes? Which may explain why it is not valid.

**Comment by ksunden** [comment]
Running locally with most recent changes and non-editable install, stubtest seems to work, with only a single unused allow list entry flagged:

```bash
$ stubtest matplotlib --mypy-config-file pyproject.toml --allowlist ci/mypy-stubtest-allowlist.txt
note: unused allowlist entry matplotlib.tests.*
Found 1 error (checked 175 modules)
```

Not fully clear to me why it behaves differently than setuptools wrt what subpackages are checked/need ignores (or in fact some of my other meson-based testing which had _three_ unused allowlist entries for subpackages, including `jpl_units` and `backends`, I believe)... but not too worried about removing such an entry. Mildly cautious that it may mean the editable and non-editable allow lists need to diverge, which would be non-ideal, but this is mostly for CI anyway.

**Comment by QuLogic** [comment]
> ```shell
> note: unused allowlist entry matplotlib.tests.*
> ```

I followed `mplsetup.cfg.template` that tests were not installed by default. However, it looks like actually only test _data_ is not installed by default (even though the option was named `package.tests`, not indicating it's for data only.) So I can fix that, and we won't have to do anything weird for stubtest.

File: lib/matplotlib/projections/meson.build [hunk 3]
```diff
@@ -0,0 +1,14 @@
+python_sources = [
+  'geo.py',
+  '__init__.py',
```
Comment (greglucas) [inline lib/matplotlib/projections/meson.build:3]: order these? They are in typing below.

File: lib/matplotlib/tests/meson.build [hunk 4]
```diff
@@ -0,0 +1,112 @@
+python_sources = [
+  'conftest.py',
+  '__init__.py',
```
Comment (greglucas) [inline lib/matplotlib/tests/meson.build:3]: reorder?

**Review by greglucas — COMMENTED** [review]
I verified this works on an intel mac with both editable and non-editable installs. The nicest thing of all was updating a compiled extension module and meson picking it up as changed and recompiling automatically even when just running an example script 🎉 so nice in an editable install to not also have to `pip install` constantly to recompile the extensions.

The one thing I did run into here was when running pytest locally. It kept warning about freetype not being 2.6.1 even though it is. I tried modifying `system_freetype` in `mplsetup.cfg` to all the different options and none of them changed anything with this warning.
```
Matplotlib is not built with the correct FreeType version to run tests.  Rebuild without setting system_freetype=1 in mplsetup.cfg.  Expect many image comparison failures below.  Expected freetype version 2.6.1.  Found freetype version 2.6.1.  Freetype build type is not local
```
I also tried adding `--config-settings=setup-args="-DrcParams-backend=Agg -Dsystem-freetype=false -Dsystem-qhull=true"` to the install and that didn't seem to do anything either.

**Comment by QuLogic** [comment]
> The one thing I did run into here was when running pytest locally. It kept warning about freetype not being 2.6.1 even though it is. I tried modifying `system_freetype` in `mplsetup.cfg` to all the different options and none of them changed anything with this warning.

I'm writing docs for this, but we won't use `mplsetup.cfg` any more.

> ```
> Matplotlib is not built with the correct FreeType version to run tests.  Rebuild without setting system_freetype=1 in mplsetup.cfg.  Expect many image comparison failures below.  Expected freetype version 2.6.1.  Found freetype version 2.6.1.  Freetype build type is not local
> ```
> 
> I also tried adding `--config-settings=setup-args="-DrcParams-backend=Agg -Dsystem-freetype=false -Dsystem-qhull=true"` to the install and that didn't seem to do anything either.

You can pass `--verbose --config-settings=compile-args=-v` to print both configure and compile time logs to see what's going on there.

**Comment by QuLogic** [comment]
I've made a few tweaks: fixed the ordering @greglucas noted, removed `certifi` since we don't do downloads ourselves, added Meson and meson-python to the minimum version CI, fixed stubtest using a non-editable install, fixed tests to install Python files but not data by default, and bumped minimum Meson to 1.1 (as the minimum version showed that the `pybind11` lookup didn't work). Finally, I've added some docs about the change.

File: doc/api/next_api_changes/development/26621-ES.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
+Build system ported to Meson
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The build system of Matplotlib has been ported from setuptools to `meson-python
+<https://meson-python.readthedocs.io>`_ and `Meson <https://mesonbuild.com>`_.
+Consequently, there has been a few changes development and packaging purposes.
+
+1. Installation by ``pip`` of packages with ``pyproject.toml`` use `build isolation
```
Comment (greglucas) [inline doc/api/next_api_changes/development/26621-ES.rst:8]: I went down the rabbit hole reading this thread:
https://github.com/pypa/pip/issues/11440
I was hoping that I could `pip install -e .[dev]` to get the build dependencies that you put into optional dev, but that doesn't appear to be the case. So is the [dev] dependency group even useful currently?

Giving users some quick commands to install build dependencies would be nice, so perhaps adding those as a requirements-dev.txt file instead and indicating we need to install from there first?

File: doc/devel/dependencies.rst [hunk 6]
```diff
@@ -223,12 +214,9 @@ Dependencies for building Matplotlib
 Setup dependencies
 ------------------
 
-- `certifi <https://pypi.org/project/certifi/>`_ (>= 2020.06.20).  Used while
-  downloading the freetype and QHull source during build.  This is not a
-  runtime dependency.
+- `meson-python <https://meson-python.readthedocs.io/>`_ (>= 0.13.1).
```
Comment (greglucas) [inline doc/devel/dependencies.rst:217]: ninja? or down below near the compiler information?

**Review by greglucas — COMMENTED** [review]
I tried to test a few different configurations locally and left some inline notes for possible fixes for the hiccups I ran into.

File: pyproject.toml [hunk 7]
```diff
@@ -1,13 +1,87 @@
...
+"Forum" = "https://discourse.matplotlib.org/"
+"Donate" = "https://numfocus.org/donate-to-matplotlib"
+
 [build-system]
-build-backend = "setuptools.build_meta"
+build-backend = "mesonpy"
+# Also keep in sync with optional dependencies above.
 requires = [
-    "certifi>=2020.06.20",
+    "meson-python>=0.13.1",
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply pyproject.toml:69]: It should be installed automatically somehow, but I think only with build isolation enabled. For an editable install (with build isolation disabled), you would need it as a development dependency. I will update those docs to make that explicit.

File: src/meson.build [hunk 8]
```diff
@@ -0,0 +1,152 @@
+# NumPy include directory - needed in all submodules
+incdir_numpy = run_command(py3,
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply src/meson.build:2]: Yes, I forgot to update that, should be easy enough to fix. I sometimes use venv-in-source, but not for this branch at the moment.

File: extern/meson.build [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  endif
+
+  freetype_proj = subproject(
+    f'freetype-@LOCAL_FREETYPE_VERSION@',
+    default_options: ['default_library=static'])
+  freetype_dep = freetype_proj.get_variable('freetype_dep')
+endif
+
+if get_option('system-qhull')
+  cc.check_header('libqhull_r/qhull_ra.h', required: true)
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply extern/meson.build:33]: Qhull was not very good at providing a pkgconfig file in the past; I will have to check if they are doing that correctly now.

File: doc/api/next_api_changes/development/26621-ES.rst (hunk 5 above)
Comment (QuLogic) [reply doc/api/next_api_changes/development/26621-ES.rst:8]: Yea, it does seem like a problem as the extra's dependencies are installed after the package itself is built. Putting it in a `requirements-dev.txt` might be the most convenient.

**Comment by QuLogic** [comment]
OK, I think I've handled the things you've found. I also found a few more cases of outdated documentation that was missed earlier.

File: pyproject.toml [hunk 10]
```diff
@@ -1,13 +1,87 @@
...
+    "Development Status :: 5 - Production/Stable",
+    "Framework :: Matplotlib",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: Education",
+    "License :: OSI Approved :: Python Software Foundation License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
```
Comment (greglucas) [inline pyproject.toml:21]:
```suggestion
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
```

File: doc/api/next_api_changes/development/26621-ES.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+      [rc_options]
+      backend=Agg
+
+      [libs]
+      system_qhull = True
+
+   may be replaced by passing the following argument to ``pip``::
+
+      --config-settings=setup-args="-DrcParams-backend=Agg -Dsystem-qhull=true"
```
Comment (greglucas) [inline doc/api/next_api_changes/development/26621-ES.rst:29]: See my comment below about this not working for me as expected.

**Review by greglucas — APPROVED** [review]
The latest changes look good to me and address the issues I was running into. I have one more small comment/question about how to use multiple setup options, but this is good to go I think.

**Comment by oscargus** [comment]
Restarted the Cygwin test to see that the same problems as in #26800 show up here as well.

File: doc/devel/dependencies.rst [hunk 12]
```diff
@@ -113,24 +114,15 @@ defaults to the system version of FreeType on AIX.
...
 
 .. code-block:: sh
 
-   python -m pip install .
-
-
-You can also use the :envvar:`MPLSETUPCFG` to specify the path to a cfg file when
-installing from pypi.
+   python -m pip install \
+     --config-settings=setup-args="-Dsystem-freetype=true -Dsystem-qhull=true" \
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/devel/dependencies.rst:124]: Did the latter actually work? Meson-python docs say there is a bug in pip that breaks that.

[the remark this answers is not in the document]
Comment (greglucas) [reply doc/devel/dependencies.rst:124]: Yes, the way you listed here does not work for me locally, but the comment I put in does work locally. I am using a zsh shell on an intel mac if that would matter at all for any of the parsing.

**Comment by ksunden** [comment]
Looks like there are problems RE: setuptools_scm not being installed on Azure and Circle

**Comment by QuLogic** [comment]
Yes, I realized it was a hard requirement, but we need to move that to development-only as in the old `setuptools` build. I'll need to update the CI to use the `dev` extra and make the dependency more optional.

**Comment by ksunden** [comment]
The cygwin failure does appear to be different than what we've been seeing on main, though it is actually with installing numpy from source, not ourselves... That said, it looks like we are missing BLAS, so might be good to sort that out...

There was an azure build failure, but looked to only be a network interruption, so reran

**Comment by QuLogic** [comment]
[BLAS is failing on `main`](https://github.com/matplotlib/matplotlib/actions/runs/6241298464/job/16943205290) too?

**Comment by ksunden** [comment]
Ah, sorry, it was a different error than the last one I looked at closely for failing cygwin... which was later on in compilation of mpl itself and seemed to have to do with random state

**Comment by oscargus** [comment]
I looked into that for #26800 and the closest I found was https://stackoverflow.com/questions/73659464/does-openblas-devel-exist-for-cygwin so I guess one will have to install 1.26 explicitly and then run with --no-build-isolation. Haven't tested it though (but a few other things).

So something like
```
pip install numpy -Csetup-args=-Dblas=blas -Csetup-args=-Dlapack=lapack
```
may work.

However, skipping build isolation, maybe there is no need to install 1.26 (yet), but stay with 1.25...

**Comment by QuLogic** [comment]
That looks like maybe what is done upstream https://github.com/numpy/numpy/pull/24523; build isolation is already disabled here because it's needed for editable Meson installs.

**Comment by QuLogic** [comment]
What's confusing to me is that we only requires `numpy!=1.21.*`, but NumPy is already installed from the Cygwin package manager at 1.21.4. So we could avoid the whole build if we weren't blocking the version that's in Cygwin, but I don't know why that's blocked?

**Comment by oscargus** [comment]
> build isolation is already disabled here because it's needed for editable Meson installs.

Ahh, sorry I missed that. Same error as I got in #26800 so I must have missed where it happened.

Here are some excerpts from the commit message for https://github.com/matplotlib/matplotlib/pull/22999/commits/e5e0e2fde61e62f56529d6d7d4ff1fa004343026

```
CI: Pin Cygwin setuptools<60 and NumPy !=1.21

NumPy 1.21 has intermittent segfaults on GHA runners, so let's just skip that.

NumPy==1.21.* has intermittent segfaults due to SIMD attempts on unaligned memory.
```

**Comment by QuLogic** [comment]
> Here are some excerpts from the commit message for [e5e0e2f](https://github.com/matplotlib/matplotlib/commit/e5e0e2fde61e62f56529d6d7d4ff1fa004343026)

Hmm, interesting and a bit weird, since those were Cygwin-built packages.

I've cherry-picked your pinning from #26800 to see what's the status here.

File: doc/devel/dependencies.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/devel/dependencies.rst:124]: I did some more testing locally, and it seems that you're right that it doesn't work. I'm not sure how or why Cygwin continued to work. The bug from the meson-python docs has been fixed in pip 23.1, so I am able to pass two arguments and have it correctly set both. I will update the documentation to note this.

[the remark this answers is not in the document]
Comment (QuLogic) [reply doc/devel/dependencies.rst]: > But it does look like you are using this style on the cygwin build and that is working...

Also, this didn't really work; the end of the configuration showed that it was actually getting `rcParams-backend: Agg -Dsystem-freetype=...`

File: .github/workflows/cygwin.yml [hunk 13]
```diff
@@ -203,20 +203,14 @@ jobs:
...
-          [libs]
-          system_freetype = False
-          system_qhull = True
-          EOT
-          cat mplsetup.cfg
           # All dependencies must have been pre-installed, so that the minver
           # constraints are held.
-          python -m pip install --no-deps -ve .
+          python -m pip install --no-deps --no-build-isolation --verbose \
+            --config-settings=setup-args="-DrcParams-backend=Agg" \
```
Comment (QuLogic) [inline .github/workflows/cygwin.yml:212]: Note that in the long commit message of e5e0e2fde61e62f56529d6d7d4ff1fa004343026, it was noted that system freetype/qhull were just for quicker testing and should probably have been reverted, so I have not carried that over to here.

**Review by oscargus — APPROVED** [review]
This seems to be OK now? To some extent it may be better to try and get this in so that there is ample time before the 3.9-release to find any issues, but I am still a bit worried to merge it myself...

Well, what is the worst that can happen?

**Comment by greglucas** [comment]
> To some extent it may be better to try and get this in so that there is ample time before the 3.9-release to find any issues, but I am still a bit worried to merge it myself...

> Well, what is the worst that can happen?

I would not be worried. You had another approving review and merged it as the maintainer's workflow says. As you say, this will give a good opportunity to see if anything does crop up rather than waiting until the last minute.

**Comment by QuLogic** [comment]
This removes special casing for AIX and IBM i OS, but the FreeType build is using an included Meson system instead of their own old autotools-based build.

@ayappanec @zheddie can you confirm whether the `main` branch still builds for you?

**Comment by ayappanec** [comment]
@KamathForAIX Can you confirm whether the main branch builds fine in AIX ?

**Comment by KamathForAIX** [comment]
Hi @QuLogic ,

So in AIX gcc with LTO option is not supported. So in meson.build file we should not use it as a default option if the target is AIX. If we can write a logic like that it would be great. 

Secondly, in the matplotlib/tools/generate_matplotlibrc.py I had to change  !/usr/bin/env python to /usr/bin/env python3.9.. 

With the two changes mentioned above we are able to build in AIX. Kindly let me know what we can do regarding the same.

**Comment by KamathForAIX** [comment]
@tacaswell Yes it works

**Comment by QuLogic** [comment]
> So in AIX gcc with LTO option is not supported. So in meson.build file we should not use it as a default option if the target is AIX. If we can write a logic like that it would be great.

If that is not supported, then this should be handled upstream in Meson. If they silently ignore `b_lto=true`, then we will have to do nothing. If they error out, then we should see if we can conditionalize it.

**Comment by KamathForAIX** [comment]
Hi @QuLogic ,

Sure. We will work this out with the meson community members. No worries. Thank you for connecting with AIX for the meson build system test for Matplotlib. 

Is there anything else we can do for Matplotlib community from AIX? Kindly let us know. 

Have a nice day ahead. 

Regards,
Aditya Kamath.

**Comment by tacaswell** [comment]
@KamathForAIX I do not know of any CI service that offer AIX workers so we are a bit blind to if we accidentally break anything.  If you could have a monthly cron-job that tries to build Matplotlib and run the tests and report any failures up to us that would be super helpful!

**Comment by KamathForAIX** [comment]
Hi @tacaswell 

Sure, we will set a CI for matplotlib and run the tests.
